libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -302,6 +306,24 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 			")\n";
 		break;
 	}
+	case FunctionType::Kind::Assert:
+	case FunctionType::Kind::Require:
+	{
+		solAssert(arguments.size() > 0, "Expected at least one parameter for require/assert");
+		solAssert(arguments.size() <= 2, "Expected at no more than two parameters for require/assert");
+
+		auto requireOrAssertFunction = m_utils.requireOrAssertFunction(
+			functionType->kind() == FunctionType::Kind::Assert,
+			arguments.size() > 1 ? arguments[1]->annotation().type : nullptr
+		);
+
+		m_code << requireOrAssertFunction << "(" << m_context.variable(*arguments[0]);
```

Comment:
```suggestion
		m_code << move(requireOrAssertFunction) << "(" << m_context.variable(*arguments[0]);
```

Reply:
Another option would be to not store it in a variable in the first place.

Reply:
Isn't this a pretty easy task for the basic optimizer in any modern c++ compiler?

Reply:
It might, but it also might not. You are telling the compiler here to perform a copy, so it will allocate new memory. The optimizer might or might not be able to remove that allocation.